src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3460,13 +3479,14 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
                               __const void *__optval, socklen_t __optlen)
 {
 	//todo check optlen and set proper errno on failure
+	si_tcp_logfunc("level=%d, optname=%d", __level, __optname);
 
 	int val, ret = 0;
 	bool supported = true;
 	bool allow_privileged_sock_opt = false;
 
 	if (0 == sockinfo::setsockopt(__level, __optname, __optval, __optlen)) {
-		return 0;
+		goto success;
```

Comment:
I think that this logic should be an issue for `SO_VMA_USER_DATA`. There is a flow when `setsockopt_kernel()` will be called and return with negative result.

Reply:
i agree